### PR TITLE
Gallery example of acquiring and overplotting a HEK polygon

### PR DESCRIPTION
**Pull request by wafels**
Queries the HEK for coronal hole data, and overplots the polygon on an AIA map.

### CODE REVIEW

File: examples/overplot_hek_polygon.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+import sunpy.map
+import sunpy.data.sample
+from sunpy.net import hek
+from sunpy.time import parse_time
+from sunpy.physics.solar_rotation import rot_hpc
+
+##############################################################################
+# Load in an AIA map
+aia_map = sunpy.map.Map(sunpy.data.sample.AIA171IMAGE)
```
Comment (Cadair) [inline examples/overplot_hek_polygon.py]: missing underscores: `AIA_171_IMAGE`

File: examples/overplot_hek_polygon.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                         ch_date,
+                         aia_map.date)
+    rotated_boundary[i, 0] = new_coords[0].value
+    rotated_boundary[i, 1] = new_coords[1].value
+
+##############################################################################
+# Now let's plot the rotated coronal hole boundary on the AIA map, and fill
+# it with some matplotlib hatching.
+fig = plt.figure()
+ax = plt.subplot()
```
Comment (Cadair) [inline examples/overplot_hek_polygon.py]: why no WCSAxes?! :(

File: examples/overplot_hek_polygon.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+aia_map.plot()
+rect = patches.Polygon(rotated_boundary, color='white', fill=False, hatch='*')
+ax.set_title('My customized plot')
```
Comment (Cadair) [inline examples/overplot_hek_polygon.py]: can we not do better than this.

**Comment by ehsteve** [comment]
Include plot in description please!

**Comment by Cadair** [comment]
@wafels I think you need a line break after the title: https://travis-ci.org/sunpy/sunpy/jobs/131441889#L1208

**Comment by wafels** [comment]
I don't understand this comment. You want me to describe the plot i am
going to make at the top of the file?

**Comment by ayshih** [comment]
He's asking you to include the plot in the PR description.

**Comment by wafels** [comment]
aia image with spoca coronal hole

**Comment by wafels** [comment]
aia image with spoca coronal hole

**Comment by Cadair** [comment]
@wafels is there a specific reason why you didn't use wcsaxes for this example?

**Comment by wafels** [comment]
The reason I am not using WCSaxes is because it is not required in order to
use SunPy.  I have an issue with the use of WCSaxes in SunPy and in the
docs.  There is nothing in the SunPy docs that says why someone should use
WCSaxes.  Some of the map plots on the sunpy.org website seem to make use
of WCSaxes, but it is not explicitly said in the docs accompanying the plot
(e.g., http://docs.sunpy.org/en/stable/guide/plotting-3.png).  I think it
might be confusing to some users when their plot does not look like the one
online simply because they have not installed WCSaxes.

**Comment by Cadair** [comment]
fair enough, that's as good of a reason as any. I think we should look at making WCSAxes a required dependancy at some point. It of course might get merged into core astropy, which makes that a lot easier.

**Comment by Cadair** [comment]
:+1:

**Comment by Cadair** [comment]
@wafels there is this section of the docs: http://docs.sunpy.org/en/stable/guide/plotting.html#plotting-maps-with-wcsaxes

What would you say we should improve?

**Pull request merged**